### PR TITLE
[SPARK-49073][SQL][DOCS] Improve the doc `sql-ref-syntax-qry-select.md`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to update the explanation for `star_clause` in the doc `sql-ref-syntax-qry-select.md`.

### Why are the changes needed?
- After the pr https://github.com/apache/spark/pull/44527, the `star_clause` now supports `EXCEPT ..` semantics, so the original explanation of `star_clause` is outdated, let's update it.

### Does this PR introduce _any_ user-facing change?
Yes, only for docs.

### How was this patch tested?
Manually test.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
Manually check as follows:
image

File: docs/sql-ref-syntax-qry-select.md [hunk 1]
```diff
@@ -76,11 +85,9 @@ SELECT [ hints , ... ] [ ALL | DISTINCT ] { [ [ named_expression | regex_column_
 
     An expression with an assigned name. In general, it denotes a column expression.
 
-    **Syntax:** `expression [[AS] alias]`
```
Comment (panbingkun) [inline docs/sql-ref-syntax-qry-select.md:79]: I have already moved this up.

**Comment by panbingkun** [comment]
cc @stefankandic @HyukjinKwon

**Comment by panbingkun** [comment]
ref: https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-qry-select.html

**Comment by stefankandic** [comment]
I don't see these link for parameters anywhere in the spark docs, so not sure we should just add them for the select page

**Comment by panbingkun** [comment]
I understand your concern that it may result in `inconsistent` document styles. 
Perhaps we can apply the `above pattern` to all parameters of "docs/sql-ref-syntax*.md" files. 
I think this `clickable parameter` can bring `convenience` to end-users.
```
(base) ➜  docs git:(master) ✗ find . -name "sql-ref-syntax*.md" | xargs grep "### Parameters" | wc -l
      89
```
WDYT @HyukjinKwon ?

File: docs/sql-ref-syntax-qry-select.md [hunk 2]
```diff
@@ -59,7 +68,7 @@ SELECT [ hints , ... ] [ ALL | DISTINCT ] { [ [ named_expression | regex_column_
     These table expressions are allowed to be referenced later in the FROM clause. This is useful to abstract
     out repeated subquery blocks in the FROM clause and improves readability of the query.
 
-* **hints**
+* **[hints](sql-ref-syntax-qry-select-hints.html)**
```
Comment (HyukjinKwon) [inline docs/sql-ref-syntax-qry-select.md:71]: If we go ahead with this, we should probably do it for all? I would separate the PR

Comment (panbingkun) [reply docs/sql-ref-syntax-qry-select.md:71]: Okay, 
I will restore the related `clickable parameter` first, and I will do it in another separate PR.

Comment (panbingkun) [reply docs/sql-ref-syntax-qry-select.md]: Done.

File: docs/sql-ref-syntax-qry-select.md [hunk 3]
```diff
@@ -41,14 +41,23 @@ select_statement [ { UNION | INTERSECT | EXCEPT } [ ALL | DISTINCT ] select_stat
...
     FROM { from_item [ , ... ] }
     [ PIVOT clause ]
     [ UNPIVOT clause ]
     [ LATERAL VIEW clause ] [ ... ] 
     [ WHERE boolean_expression ]
     [ GROUP BY expression [ , ... ] ]
     [ HAVING boolean_expression ]
+
+named_expression
+   expression [[AS] alias]
```
Comment (allisonwang-db) [inline docs/sql-ref-syntax-qry-select.md:54]:
```suggestion
   expression [ [ AS ] alias ]
```

File: docs/sql-ref-syntax-qry-select.md [hunk 4]
```diff
@@ -41,14 +41,23 @@ select_statement [ { UNION | INTERSECT | EXCEPT } [ ALL | DISTINCT ] select_stat
 
 While `select_statement` is defined as
 ```sql
-SELECT [ hints , ... ] [ ALL | DISTINCT ] { [ [ named_expression | regex_column_names | star ] [ , ... ] | TRANSFORM (...) ] }
+SELECT [ hints , ... ] [ ALL | DISTINCT ] { [ [ named_expression | regex_column_names | star_clause ] [ , ... ] | TRANSFORM (...) ] }
```
Comment (allisonwang-db) [inline docs/sql-ref-syntax-qry-select.md:44]: I wonder if we should do `star [ except_clause ]` instead of having except_clause inside the star_clause.

Comment (panbingkun) [reply docs/sql-ref-syntax-qry-select.md:44]: I am referring to this doc https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-qry-select.html, as follows:
image
Based on my understanding, I think `one possible reason` is that `EXCEPT` cannot exist independently of `STAR`, which seems to emphasize this relationship.

**Pull request closed without merging**